dependency/client_set.go

```diff
@@ -135,7 +137,11 @@ func (c *ClientSet) CreateConsulClient(i *CreateConsulClientInput) error {
 		tlsConfig.BuildNameToCertificate()
 
 		// SSL verification
-		if !i.SSLVerify {
+		if i.ServerName != "" {
+			tlsConfig.ServerName = i.ServerName
+			tlsConfig.InsecureSkipVerify = false
+			log.Printf("[DEBUG] (clients) using explicit consul TLS server host name: %s", tlsConfig.ServerName)
+		} else if !i.SSLVerify {
```

Comment:
I think this logic (and the logic on 222) could lead to some subtle challenges. Basically if you set a server name, you can _never_ disable SSL verification. In essence this breaks the ability to have a later configuration that is merged (remember CT can merge multiple configs) that would disable SSL verification.

A use case I'm thinking is that a company has a default "base" config, but then they disable SSL verification when running in local development. As written, this change would require them to both disable SSL verification AND set the server_name to `""`.